packages/builder/src/components/common/bindings/DrawerBindableInput.svelte

```diff
@@ -23,18 +23,25 @@
   const dispatch = createEventDispatcher()
   let bindingDrawer
   let valid = true
+  let currentVal = value
 
   $: readableValue = runtimeToReadableBinding(bindings, value)
   $: tempValue = readableValue
   $: isJS = isJSBinding(value)
 
   const saveBinding = () => {
     onChange(tempValue)
+    onBlur()
     bindingDrawer.hide()
   }
 
   const onChange = value => {
-    dispatch("change", readableToRuntimeBinding(bindings, value))
+    currentVal = readableToRuntimeBinding(bindings, value)
+    dispatch("change", currentVal)
+  }
+
+  const onBlur = () => {
+    dispatch("blur", currentVal)
```

Comment:
Could you quickly explain this change? It looks like it's for the key value builder below but I wonder if there is some other bug at work if using the normal `on:change` handlers wasn't working.

Reply:
`on:change` does work - but I wanted to use blur instead of change in the REST UI, as I'm parsing the end result - it involves bi-directional parsing and getting it working with on change would have been a nightmare, keeping track of the state + re-parsing constantly didn't make sense, it felt better to wait until the user input was complete.